core/include/vecmem/memory/instrumenting_memory_resource.hpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include <cstddef>
+#include <functional>
+
+#include "vecmem/memory/details/memory_resource_base.hpp"
+#include "vecmem/memory/memory_resource.hpp"
```

Comment:
I think it would be healthy to add an explicit include for the auto-generated `vecmem/vecmem_core_export.hpp`. `memory_resource_base.hpp` pulls it in, but we shouldn't rely on that.

Reply:
Added!